app/javascript/components/subject/CaseStatus.js

```diff
@@ -102,8 +106,8 @@ class CaseStatus extends React.Component {
   createModal(title, toggle, submit) {
     if (
       this.props.patient.isolation &&
-      (this.props.patient.case_status === 'Confirmed' || this.props.patient.case_status === 'Probable') &&
-      (this.state.case_status === 'Suspect' || this.state.case_status === 'Unknown' || this.state.case_status === 'Not a Case' || this.state.case_status === '')
+      !this.state.confirmedOrProbable &&
+      (this.props.patient.case_status === 'Confirmed' || this.props.patient.case_status === 'Probable')
```

Comment:
You can remove `(this.props.patient.case_status === 'Confirmed' || this.props.patient.case_status === 'Probable')` now. This if-statement can just be:
`if (this.props.patient.isolation && this.state.confirmedOrProbable)`

Reply:
this conditional is comparing what the case status was to what the user changed it too (a very specific case for the modal text), hence why it is the case status on `this.props.patient` not `this.state`. This needs to stay as is

Reply:
Ah missed that it was on the props rather than the state. All set then.